content/ja/docs/concepts/services-networking/cluster-ip-allocation.md

```diff
@@ -0,0 +1,126 @@
+---
+title: ServiceのClusterIPの割り当て
+content_type: concept
+weight: 120
+---
+
+
+<!-- overview -->
+
+Kubernetesでは、[Service](/ja/docs/concepts/services-networking/service/)はPodとして実行しているアプリケーションを抽象的に公開する方法です。Serviceはクラスター内で仮想IPアドレス(type: ClusterIPのServiceを使用)を持つことができます。クライアントはその仮想IPアドレスを使用してServiceに接続することができます。KubernetesはそのServiceを通して複数のPodにトラフィックを負荷分散します。
+
+<!-- body -->
+
+## どうやってServiceのClusterIPは割り当てられるのか？
+
+KubernetesがServiceに仮想IPアドレスを割り当てる必要がある場合、2つの方法の内どちらかの方法で行われます:
+
+_動的割り当て_
+: クラスターのコントロールプレーンは自動的に`type: ClusterIP` のServiceのために設定されたIP範囲の中から未割り当てのIPアドレスを選びます。
+
+_静的割り当て_
+: Serviceのために設定されたIP範囲の中から自身でIPアドレスを選びます。
+
+クラスター全体を通して、Serviceの`ClusterIP`はユニークでなければいけません。割り当て済みの`ClusterIP`を使用してServiceを作成しようとするとエラーが返ってきます。
+
+## なぜServiceのClusterIPを予約する必要があるのか？
+
+時には、クラスター内の他のコンポーネントやユーザーが利用できるように、Serviceをよく知られたIPアドレスで実行したい場合があります。
+
+良い例はクラスターのDNSサービスになります。慣習として、一部のKubernetesインストーラーはServiceのIP範囲の10番目のIPアドレスをDNSサービスに割り当てます。ServiceのIP範囲を10.96.0.0/16とするクラスターを構成し、DNSサービスのIPを10.96.0.10にするとします。この場合、下記のようなServiceを作成する必要があります。
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    k8s-app: kube-dns
+    kubernetes.io/cluster-service: "true"
+    kubernetes.io/name: CoreDNS
+  name: kube-dns
+  namespace: kube-system
+spec:
+  clusterIP: 10.96.0.10
+  ports:
+  - name: dns
+    port: 53
+    protocol: UDP
+    targetPort: 53
+  - name: dns-tcp
+    port: 53
+    protocol: TCP
+    targetPort: 53
+  selector:
+    k8s-app: kube-dns
+  type: ClusterIP
+```
+
+しかし、前述したように10.96.0.10のIPアドレスは予約されていません。動的割り当てを使用して、他のServiceが先にまたは同時に作成された場合、そのIPアドレスが割り当てられる可能性があります。その場合、競合エラーで失敗しDNSサービスを作成することができません。
```

Comment:
```suggestion
しかし、前述したように10.96.0.10のIPアドレスは予約されていません。他のServiceが動的割り当てよりも前に、または同時に作成された場合、このIPアドレスがそのServiceに割り当てられる可能性があります。その場合、競合エラーで失敗しDNS Serviceを作成することができません。
```
より状況が伝わりやすくなるようにしてみました。
> but as it was explained before, the IP address 10.96.0.10 has not been reserved; if other Services are created before or in parallel with dynamic allocation, there is a chance they can allocate this IP, hence, you will not be able to create the DNS Service because it will fail with a conflict error.